dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTaskTest.java

```diff
@@ -0,0 +1,345 @@
+
+            Assert.assertNotNull(columns);
+
+            Assert.assertTrue(columns.length == 2);
+
+            Assert.assertEquals(Arrays.toString(columns), "[`a`, `table`]");
+        }
+        catch (Exception e) {
+            throw e;
+        }
+    }
+
+    /**
+     * Method: tryGrammaticalParsingSqlColumnNames(DbType dbType, String sql)
+     */
+    @Test
+    public void testTryGrammaticalAnalysisSqlColumnNames()
+        throws Exception {
+        try {
+            Method method = DataxTask.class.getDeclaredMethod("tryGrammaticalAnalysisSqlColumnNames", DbType.class, String.class);
+            method.setAccessible(true);
+            String[] columns = (String[]) method.invoke(dataxTask, DbType.MYSQL, "select t1.a, t1.b from test t1 union all select a, t2.b from (select a, b from test) t2");
+
+            Assert.assertNotNull(columns);
+
+            Assert.assertTrue(columns.length == 2);
+
+            Assert.assertEquals(Arrays.toString(columns), "[a, b]");
+        }
+        catch (Exception e) {
+            throw e;
+        }
+    }
+
+    /**
+     * Method: tryExecuteSqlResolveColumnNames(BaseDataSource baseDataSource,
+     * String sql)
+     */
+    @Test
+    public void testTryExecuteSqlResolveColumnNames()
+        throws Exception {
+        // TODO: Test goes here...
+    }
+
+    /**
+     * Method: buildDataxJsonFile()
+     */
+    @Test
+    public void testBuildDataxJsonFile()
+        throws Exception {
+        try {
+            Method method = DataxTask.class.getDeclaredMethod("buildDataxJsonFile");
+            method.setAccessible(true);
+            String filePath = (String) method.invoke(dataxTask, null);
+            Assert.assertNotNull(filePath);
+        }
+        catch (Exception e) {
+            throw e;
+        }
+    }
+
+    /**
+     * Method: buildDataxJobContentJson()
+     */
+    @Test
+    public void testBuildDataxJobContentJson()
+        throws Exception {
+        try {
+            Method method = DataxTask.class.getDeclaredMethod("buildDataxJobContentJson");
+            method.setAccessible(true);
+            List<JSONObject> contentList = (List<JSONObject>) method.invoke(dataxTask, null);
+            Assert.assertNotNull(contentList);
+
+            JSONObject content = contentList.get(0);
+            JSONObject reader = (JSONObject) content.get("reader");
+            Assert.assertNotNull(reader);
+
+            String readerPluginName = (String) reader.get("name");
+            Assert.assertEquals(readerPluginName, DataxUtils.DATAX_READER_PLUGIN_MYSQL);
+
+            JSONObject writer = (JSONObject) content.get("writer");
+            Assert.assertNotNull(writer);
+
+            String writerPluginName = (String) writer.get("name");
+            Assert.assertEquals(writerPluginName, DataxUtils.DATAX_WRITER_PLUGIN_MYSQL);
```

Comment:
assertEquals first param is expected value, sceond param is actual value, please swap there params.